packages/builder/src/pages/builder/workspace/[application]/agent/[agentId]/AgentToolConfigModal.svelte

```diff
@@ -190,75 +291,79 @@
             <Body size="S" color="var(--spectrum-global-color-gray-700)">
               {source.description}
             </Body>
-            {#if $agentsStore.toolSources.some(ts => ts.type === source.type)}
-              <div class="connected-badge">
-                <Icon name="CheckmarkCircle" size="S" />
-                Connected
-              </div>
-            {/if}
           </div>
         {/each}
       </div>
     {:else if mode === "configure" && selectedSourceType}
       <div class="config-form">
-        <div class="auth-section">
-          <Heading size="XS">Authentication</Heading>
-          {#if selectedSourceType.type === "GITHUB"}
-            <Input
-              label="API Key"
-              bind:value={config.apiKey}
-              type="password"
-              placeholder="Enter your GitHub API token"
-            />
-            <Input
-              label="Base URL (Optional)"
-              bind:value={config.baseUrl}
-              type="text"
-              placeholder="https://api.github.com (leave empty for default)"
-            />
-          {:else if selectedSourceType.type === "CONFLUENCE"}
-            <Input
-              label="API Key"
-              bind:value={config.apiKey}
-              type="password"
-              placeholder="Enter your Confluence API token"
-            />
-            <Input
-              label="Email Address"
-              bind:value={config.email}
-              type="email"
-              placeholder="your.email@domain.com"
-              helpText="Your Atlassian account email address"
-            />
-            <Input
-              label="Base URL (Optional)"
-              bind:value={config.baseUrl}
-              type="text"
-              placeholder="https://your-domain.atlassian.net"
-            />
-          {:else if selectedSourceType.type === "BAMBOOHR"}
-            <Input
-              label="API Key"
-              bind:value={config.apiKey}
-              type="password"
-              placeholder="Enter your BambooHR API key"
-            />
-            <Input
-              label="Subdomain"
-              bind:value={config.subdomain}
-              type="text"
-              placeholder="your-company"
-              helpText="Your BambooHR subdomain (e.g., 'mycompany' for mycompany.bamboohr.com)"
-            />
-          {/if}
+        {#if selectedSourceType.type === ToolSourceType.REST_QUERY}
+          <div class="rest-query-section">
```

Comment:
NAB: Maybe move this section into its own component? `RestQueryToolConfig` or something like that.